activesupport/test/core_ext/object/inclusion_test.rb

```diff
@@ -32,6 +33,17 @@ def test_in_set
     assert_not 3.in?(s)
   end
 
+  def test_in_date_range
+    assert Date.today.in?(Date.yesterday..Date.tomorrow)
+    assert Date.today.in?(Date.yesterday..)
+
+    assert Date.today.in?(..Date.tomorrow)
+    assert_not Date.today.in?(..Date.yesterday)
+
+    # This never returns in Ruby 3.1
+    Timeout.timeout(0.1) { assert_not Date.today.in?(Date.tomorrow..) }
```

Comment:
I think we can remove this assertion. We don't need to test all the behavior of `cover?`. This will avoid the need for this test to wait at least 0.1s to finish.

Reply:
Sounds good, also,is no longer necessary for 3.1,  I had it before introducing cover in Object#in? change for the test to fail fast.

Reply:
Removed the timeout from last assertion, also reworded the method documentation a bit.